pybv/io.py

```diff
@@ -281,6 +307,102 @@ def write_brainvision(*, data, sfreq, ch_names,
+        try:
+            events.astype(float)
+        except ValueError:
+            raise ValueError("When array, events must be numeric, but found non-numeric types")
+
+    # validate input: list of dict
+    assert isinstance(events, list)  # must be true
+    for event in list:
+
+        # each item must be dict
+        if not isinstance(event, dict):
+            raise ValueError("When list, events must be a list of dict, but found non-dict element in list")
+
+        # required keys
+        for required_key in ["onset", "description"]:
+            if required_key not in event:
+                raise ValueError("When list of dict, each dict in events must have the keys 'onset' and 'description'")
+
+        # populate keys with default if missing (in-place)
+        # NOTE: using "ch_names" as default for channels translates directly
+        #       into "all" but is robust with respect to channels named
+        #       "all"
+        event_defaults = dict(duration=1, type="Stimulus", channels=ch_names)
+        for optional_key, default in event_defaults.items():
+            event[optional_key] = event.get(optional_key, default)
+
+        # validate key types
+        # `onset`, `duration`
+        for key in ["onset", "duration"]:
+            if not isinstance(event[key], int):
+                raise ValueError(f"events: `{key}` must be int")
+
+        # `type`
+        event_types = ["Stimulus", "Response", "Comment"]
+        if event["type"] not in event_types:
+            raise ValueError(f"events: `type` must be one of {event_types}")
+
+        # `description`
+        if event["type"] in ["Stimulus", "Response"]:
+            if not isinstance(event["description"], int):
+                raise ValueError(f"events: when `type` is {event['type']}, `description` must be int")
+        else:
+            assert event["type"] == "Comment"
+            if not isinstance(event["description"], (int, str)):
+                raise ValueError(f"events: when `type` is {event['type']}, `description` must be str or int")
+
+        # `channels`
+        # "all" becomes ch_names (list of all channel names)
+        # single str 'ch_name' becomes [ch_name]
+        if not isinstance(event["channels"], (list, str)):
+            raise ValueError("events: `channels` must be str or list of str")
+
+        if isinstance(event["channels"], str):
+            if event["channels"] == "all":
+                if "all" in ch_names:
+                    raise ValueError("Found channel named 'all'. Your `channels` specification in events is also 'all': This is ambiguous, because 'all' is a reserved keyword. Either rename the channel called 'all', or explicitly list all ch_names in `channels` in each event instead of using 'all'")
+                event["channels"] = ch_names
+            else:
+                event["channels"] = [event["channels"]]
+
```

Comment:
also: raise a warning if channels is anything other than "all" or a single channel

Reply:
done